### ISSUE TITLE
Opening a folder from desktop invalidate previous saved tabs

### ISSUE THREAD

**Issue by redtide** [lxqt/pcmanfm-qt#1517]
I use PCMFM in single window mode, set in Preferences -> Behavior -> Single window mode.
(hope it will be clear even if the english strings are different; for some strange reason I'm not able to run
`LANG=en pcmanfm-qt` from terminal) plus reopen tabs from previous session (Preferences -> UI -> last checkmark to reopen previous tabs in a new window).
This works until I open a folder from the desktop, in which case it discard the previous session saved tabs.

##### Expected Behavior

Opening any folder, even from desktop, should be added to previous saved tabs from previous sessions.

##### Current Behavior

Open any folder from the desktop, including places like Computer, Network etc.
makes lost all previous folder tabs, which are lost also on new sessions.

##### Steps to Reproduce (for bugs)

1. Open some folders with the same settings I described
2. Close PCManFM-Qt
3. Open a folder from the desktop

##### Context

It's important to me to keep/save all project folders open I work with.

##### System Information

* PcManFm-qt version: 1.0.0-1
* Distribution & Version: Arch
* Qt Version: 5.15.2+kde+r286-1
* liblxqt Version: 1.0.0-1

**Comment by tsujan** [comment]
I'm afraid your interpretation of "Reopen last window tabs in a new window" is incorrect and this isn't related to desktop.

The option in question means that *an empty session* starts with the tabs of the last session. If you close all windows and open a folder (from command-line, or by clicking a local link, or by clicking a desktop item, or...), you've started a *nonempty* session with that folder; therefore, the tabs list should be reset. It would be very confusing if the tabs list was accumulated indefinitely.

In short, the behavior is intended; hence closing this issue.

Whether the text of the option is clear enough or not is another question. Perhaps we could have said "Reopen last window tabs with an empty session" but, IMHO, it would be too technical and most users expect nothing other than that.

**Comment by tsujan** [comment]
"Reopen last window tabs with an empty session" would have another problem too: technically, "session" means pcmanfm-qt's process, while here it only means "a new window without specifying a folder". I think the original text is OK. If you have a better suggestion, I'm all ears.

**Comment by redtide** [comment]
"Reopen the previous file manager session window tabs when on a new/empty session"? No idea, also because I read the Italian translation, ATM IDK how the English one is. Anyway I see that my needs are about another (eventually new) feature then, something like "Reuse always the same file manager session".

**Comment by stefonarch** [comment]
To see pcmanfm-qt in en:
* stop the "desktop module in "session settings"
* `LANG=C pcmanfm-qt`

**Comment by tsujan** [comment]
> "Reopen the previous file manager session window tabs when on a new/empty session"?

Toooo long ;)

"Reopen last window tabs on opening new window" would also be an acceptable label but I think most users would see it synonymous with the current text.

> Anyway I see that my needs are about another (eventually new) feature then, something like "Reuse always the same file manager session".

When a user opens a folder or a list of folders (e,g., in a terminal), only the folder(s) should be opened because the user has explicitly asked for that. Any other behavior would be inconsistent and should be considered a bug, not a feature.

**Comment by redtide** [comment]
> To see pcmanfm-qt in en:

Nice, thank you!

> When a user opens a folder or a list of folders (e,g., in a terminal), only the folder(s) should be opened because the user has explicitly asked for that. Any other behavior would be inconsistent and should be considered a bug, not a feature.

I don't see the problem if the user explicitly sets to reuse the old session, anyway I can just avoid to open folders from the desktop and use a new tab, thank you for explanations!

**Comment by tsujan** [comment]
> I don't see the problem if the user explicitly sets to reuse the old session

When I was implementing this feature, I had to think about all combinations, without cluttering the settings dialog. The current behavior was the most self-consistent behavior that came to my mind.

What should happen if "Single window mode" isn't checked (the default)? I open a window and it contains 5 tabs; should they be reopened if I click a Desktop folder? Without "Single window mode", the concept of session doesn't make much sense (hence talking about "last window tabs" and not "last session"), while the option in question doesn't depend on it.

On the other hand, if the behavior was made dependent on the single-window mode, another inconsistency would arise: two different behaviors for the same option. Adding yet another option? It felt like an overkill. I'm experiencing déjà vu...

**Comment by redtide** [comment]
I understand that this could over complicate the current logic of the application, OTOH I can't expect such feature to be implemented just for me if nobody else need it, just saying that this could be consistent with a web browser usage, where opening a URL desktop file would open a new tab even if the browser is closed; of course it would require both options enabled. As I said it's just about to workaround on this by avoiding to use links on the desktop and do it from within the file manager.

**Comment by stefonarch** [comment]
I had an idea, feel free to trash it: 
"sticky" tabs that always will be opened when starting it, just a hook on the tab button.

**Comment by tsujan** [comment]
> just saying that this could be consistent with a web browser usage

I wanted to say that it is *not* similar to a browser but felt that there was no need to it after the above explanations ;)

**Comment by tsujan** [comment]
@stefonarch 

It isn't that simple. https://github.com/lxqt/pcmanfm-qt/issues/1517#issuecomment-1013878037 was just a part of it. Buggy behavior will be inevitable, whether they'll be reported or not.
